SharedPackages/VPN/Sources/VPN/WireGuardKit/WireGuardAdapter.swift

```diff
@@ -211,28 +222,38 @@ public class WireGuardAdapter {
         return nil
     }
 
-    /// Returns the tunnel device interface name, or nil on error.
-    /// - Returns: String.
-    public var interfaceName: String? {
-        guard let tunnelFileDescriptor = self.tunnelFileDescriptor else { return nil }
+    /// Returns the tunnel device interface name.
+    /// - Returns: The interface name as a String.
+    /// - Throws: `WireGuardAdapterError.cannotLocateTunnelFileDescriptor` if the tunnel file descriptor is not available.
+    /// - Throws: `WireGuardAdapterError.interfaceNameBufferAllocationFailed` if buffer allocation fails.
+    /// - Throws: `WireGuardAdapterError.getInterfaceNameFailed` if the getsockopt system call fails, with the underlying POSIX error code.
+    public var interfaceName: String {
+        get throws {
+            guard let tunnelFileDescriptor = self.tunnelFileDescriptor else {
+                throw WireGuardAdapterError.cannotLocateTunnelFileDescriptor
+            }
 
-        var buffer = [UInt8](repeating: 0, count: Int(IFNAMSIZ))
+            var buffer = [UInt8](repeating: 0, count: Int(IFNAMSIZ))
 
-        return buffer.withUnsafeMutableBufferPointer { mutableBufferPointer in
-            guard let baseAddress = mutableBufferPointer.baseAddress else { return nil }
+            return try buffer.withUnsafeMutableBufferPointer { mutableBufferPointer in
+                guard let baseAddress = mutableBufferPointer.baseAddress else {
+                    throw WireGuardAdapterError.interfaceNameBufferAllocationFailed
+                }
 
-            var ifnameSize = socklen_t(IFNAMSIZ)
-            let result = getsockopt(
-                tunnelFileDescriptor,
-                2 /* SYSPROTO_CONTROL */,
-                2 /* UTUN_OPT_IFNAME */,
-                baseAddress,
-                &ifnameSize)
+                var ifnameSize = socklen_t(IFNAMSIZ)
+                let result = getsockopt(
+                    tunnelFileDescriptor,
+                    2 /* SYSPROTO_CONTROL */,
+                    2 /* UTUN_OPT_IFNAME */,
+                    baseAddress,
+                    &ifnameSize)
 
-            if result == 0 {
-                return String(cString: baseAddress)
-            } else {
-                return nil
+                if result == 0 {
+                    return String(cString: baseAddress)
+                } else {
+                    let posixError = NSError(domain: NSPOSIXErrorDomain, code: Int(errno), userInfo: nil)
+                    throw WireGuardAdapterError.getInterfaceNameFailed(posixError)
+                }
```

Comment:
This diff is the gist of the PR.  I know it looks awful in GH, but the only changes should be:
- This calculated var can now throw.
- If there's an error it'll throw the error, where it previously was just bailing out silently.

Anything else that's not this should be 1:1 unchanged.